core/src/saros/activities/FileActivity.java

```diff
@@ -49,6 +51,8 @@
    * @param content content of the file denoted by the path (only valid for {@link Type#CREATED} and
    *     {@link Type#MOVED})
    * @param encoding the encoding the content is encoded with or <code>null</code>
+   * @param fileType the file type of the new file (only used for {@link Type#CREATED} and {@link
```

Comment:
Why is this necessary for file moves ?

Reply:
When I wrote the `SharedResourcesManager` (component handling incoming file activities), I was acting under the assumption that it should mainly operate on the Saros API.

As a result, received file moves are handled through the saros IResource API, meaning the move is realized through the creation of a new resource and the deletion of the old resource. If the old file initially had no file type association, the new file won't have one either, meaning we need to set it manually.